.vscode/settings.json

```diff
@@ -0,0 +1,10 @@
+{
```

Comment:
usually, it's not recommended to add IDE specific setting files. I'm kind of opposed to add this here.

Reply:
I'll yank that file for now, but I'll add here my comments from discord:

I normally agree, but these are workspace settings. They shouldn't interfere with any user settings. We can make it part of the contribution to ask people not to change that file.

Vs code has 3 levels of settings, including a user settings so it shouldn't be a big deal for people to customize them.
I personally think .vscode folder should be committed, and instructions on why it's there added to most projects. It's a great way to share project specific settings that need to exist.
In this case, we can nuke it in favor of instructions on how to set it up and use it yourself, but I'm 100% sure use cases like this are why vscode has layers of settings. Doesn't idea?

Reply:
In this case, without the .vscode folder, devs will have to manually set up the formatter and checkstyle settings with each new clone. We can include instructions for that, but I find it a better choice to give them the settings by default in the 'Workspace' settings, and instruct them to use their 'User' or 'System' settings for personal customizations.

Reply:
I'll make it a separate PR to be discussed on, though.

Reply:
I've removed the devcontainer and vscode settings folders and will send separate PR for them.